ci/run_fuzzit.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash -eu
+# Copyright 2018 Google Inc.
+# Modifications 2019 fuzzit.dev Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# xlarge resource_class.
+# See note: https://circleci.com/docs/2.0/configuration-reference/#resource_class for why we
+# hard code this (basically due to how docker works).
+export NUM_CPUS=8
+
+export CC=clang
+export CXX=clang++
+
+cat /proc/meminfo
+
+# Disable UBSan vptr since target built with -fno-rtti.
+export CFLAGS="-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -fno-sanitize=vptr"
+export CXXFLAGS="-O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -stdlib=libc++ -fno-sanitize=vptr"
+
+declare -r FUZZER_TARGETS_CC=$(find . -name *_fuzz_test.cc)
+declare -r FUZZER_TARGETS="$(for t in ${FUZZER_TARGETS_CC}; do echo "${t:2:-3}"; done)"
+
+FUZZER_DICTIONARIES="\
+"
+
+# Copy $CFLAGS and $CXXFLAGS into Bazel command-line flags, for both
+# compilation and linking.
+#
+# Some flags, such as `-stdlib=libc++`, generate warnings if used on a C source
+# file. Since the build runs with `-Werror` this will cause it to break, so we
+# use `--conlyopt` and `--cxxopt` instead of `--copt`.
+#
+declare -r EXTRA_BAZEL_FLAGS="$(
+for f in ${CFLAGS}; do
+  echo "--conlyopt=${f}" "--linkopt=${f}"
+done
+for f in ${CXXFLAGS}; do
+  echo "--cxxopt=${f}" "--linkopt=${f}"
+done
+)"
+
+declare BAZEL_BUILD_TARGETS=""
+declare FILTERED_FUZZER_TARGETS=""
+for t in ${FUZZER_TARGETS}
+do
+  declare BAZEL_PATH="//"$(dirname "$t")":"$(basename "$t")
+  declare TAGGED=$(bazel query "attr('tags', 'no_fuzz', ${BAZEL_PATH})")
+  if [ -z "${TAGGED}" ]
+  then
+    FILTERED_FUZZER_TARGETS+="$t "
+    BAZEL_BUILD_TARGETS+="${BAZEL_PATH}_driverless "
+  fi
+done
+
+# Build driverless libraries.
+bazel build --jobs=${NUM_CPUS} --verbose_failures --dynamic_mode=off --spawn_strategy=standalone \
```

Comment:
Thanks @yevgenypats. This is pretty exciting to see coming together! If I'm correct, you manage to run the sanity in ~20 mins? That seems pretty good and should be acceptable.

I have some remaining concern over build maintenance. These files basically mirror what we have up at https://github.com/google/oss-fuzz/tree/master/projects/envoy. I'm wondering if you could refactor to better share code, otherwise we need to update both places each time we need to fix the build.

Ideas:
1. Pushing more of these build options into Bazel profiles, e.g. like the kind of thing we do in https://github.com/envoyproxy/envoy/blob/master/.bazelrc#L85.
2. Refactoring the bash files in oss-fuzz repo so you can source various functions from them here.
3. Doing more of the linking and assembly of the libfuzz driver + tests directly in a synthetic target in https://github.com/envoyproxy/envoy/blob/master/bazel/envoy_test.bzl#L64.

Some combination of these should make things much easier to manage as a permanent thing.

WDYT?

Reply:
@htuch sounds reasonable. I'll try to find the some combination of the ideas you suggested.